multinet/api/tasks/upload/csv.py

```diff
@@ -67,3 +69,129 @@ def process_csv(
 
     # Insert rows
     table.put_rows(csv_rows)
+
+
+def create_csv_network(workspace: Workspace, serializer):
+    """Create a network from a link of tables (in request thread)."""
```

Comment:
I really don't have a good metric, since I've only done it on fairly small data. However, I can say at least for the arango query, it probably has a runtime complexity of O(MxN), where M = the number of documents in the foreign table, and N = the number of documents in the source table (but I don't know how much faster this is made through arango's query optimization). It might be possible to improve this runtime slightly, but I'm not too worried about it at the moment.

Reply:
The reason I asked is that this seems like a classic use case for asynchronous processing. As long as we're ready to "port" this to a Celery task, then I'm ok with it for now.

Reply:
I agree, I just kept it simple for this initial implementation but it should be trivial to run as a celery task (hardest part would be implementing client side polling).